dune/xt/la/container/common.hh

```diff
@@ -762,49 +837,63 @@ public:
   } // CommonSparseMatrix(...)
 
   template <int ROWS, int COLS>
-  explicit operator Dune::FieldMatrix<ScalarType, ROWS, COLS>() const
+  explicit operator std::unique_ptr<Dune::FieldMatrix<ScalarType, ROWS, COLS>>() const
```

Comment:
For larger matrices (something like 1000*1000) creating a FieldMatrix on the stack causes a stack-overflow. This happened to me and just returning a unique_ptr seemed to be the fastest solution. I am not sure how else to avoid the stack-overflow. Maybe we can just add back the old operator without the unique_ptr and also keep the new one, so we can use the old operator whenever possible and explicitly use the unique_ptr one if the matrix is too large.